src/nylas-connection.ts

```diff
@@ -64,69 +61,61 @@ export default class NylasConnection {
     this.clientId = clientId;
   }
 
-  requestOptions(options: RequestOptions) {
-    const newOptions: CoreOptions & UrlOptions = {
-      method: options.method || 'GET',
-      url: `${config.apiServer}${options.path}`,
-      headers: options.headers || {},
-      json: options.json == null ? true : options.json,
-      useQuerystring: true,
-    };
-
-    if (options.downloadRequest) {
-      newOptions.encoding = null;
-    }
-
-    if (options.formData) {
-      newOptions.formData = options.formData;
-    } else {
-      newOptions.body = options.body || {};
-    }
-
-    // For convenience, If `expanded` param is provided, convert to view:
-    // 'expanded' api option
+  requestOptions(options: RequestOptions): RequestOptions {
```

Comment:
Maybe combine with getRequest method I added specifically for fetch?

Reply:
I actually addressed this but forgot to commit the code! They're very similar methods so now we have a `newRequest` method that returns the new `Request` object so that the delta class can do its own processing separate from what happens in the `request` method. Let me know what you think/comments.